pkg/schemacmp/table_test.go

```diff
@@ -518,20 +608,54 @@ func (s *tableSchema) TestJoinSchemas(c *C) {
 }
 
 func (s *tableSchema) TestTableString(c *C) {
-	ti, err := s.toTableInfo("CREATE TABLE tb (a INT, b INT)")
+	// Emtpy charset and collate shouldn't exist in Encode.String().
+	ti, err := s.toTableInfo("CREATE TABLE tb (id INT NOT NULL)")
 	c.Assert(err, IsNil)
-
 	charsets := []string{"", mysql.DefaultCharset}
 	collates := []string{"", mysql.DefaultCollationName}
 	for _, charset := range charsets {
 		for _, collate := range collates {
 			ti.Charset = charset
 			ti.Collate = collate
-			sql := strings.ToLower(schemacmp.Encode(ti).String())
+			sql := strings.ToLower(Encode(ti).String())
 			c.Assert(strings.Contains(sql, "charset"), Equals, charset != "")
 			c.Assert(strings.Contains(sql, "collate"), Equals, collate != "")
 			_, err := s.toTableInfo(sql)
 			c.Assert(err, IsNil)
 		}
 	}
+
+	// Table with partition should be parsed after encoded.
+
+	// Range partition.
+	ti, err = s.toTableInfo("CREATE TABLE tb (id INT NOT NULL) PARTITION BY RANGE (id) (PARTITION p0 VALUES LESS THAN (10), PARTITION p1 VALUES LESS THAN (20))")
+	c.Assert(err, IsNil)
+	sql := strings.ToLower(Encode(ti).String())
+	c.Assert(strings.Contains(sql, "partition by range"), Equals, true)
```

Comment:
I think above line we use ToLower to handle `charset` and `CHARSET`, here no need to use it

Reply:
here is to check whether the sql  statement contains the partition definition.

Reply:
We could remove ToLower and check PARTITION BY RANGE

Reply:
`charset` check is no need to use `ToLower` too 😂

Reply:
maybe the original auther wants to check `charset` and `CHARSET`

Reply:
I'm the original author 😂

Reply:
I originally wanted to increase the stability, because the SQL keywords are case-insensitive.